src/hooks/useContract.ts

```diff
@@ -154,6 +155,11 @@ export const useSpecialBunnyPredictionContract = () => {
   return useMemo(() => getBunnySpecialPredictionContract(library.getSigner()), [library])
 }
 
+export const useFarmAuctionContract = () => {
+  const { account, library } = useActiveWeb3React()
+  return useMemo(() => getFarmAuctionContract(account ? library.getSigner() : library), [library, account])
```

Comment:
Different from other `useXXXContract` for reasons described in slack. This "fix" makes it possible to properly interact with every part of UI, being it connected, unconnected or whitelisted user, however some console errors are present in browser console. No crashes.
Will try to spend some time debugging it again, but it's not a show stopper for release.

Reply:
what kind of error ?

Reply:
If that is the case then we should add this as a comment in the code.

Reply:
Added comment - 8667b400183cb963307de01e7c1db3f96fb87d5e